packages/block-library/src/embed/edit.native.js

```diff
@@ -190,6 +193,10 @@ const EmbedEdit = ( props ) => {
 		}
 	}, [ preview, isEditingURL ] );
 
+	useEffect( () => setShowEmbedBottomSheet( isEditingURL ), [
+		isEditingURL,
+	] );
```

Comment:
This side-effect will assure that the embed bottom sheet is shown/hidden when `isEditingURL` is updated.